app/scripts/controllers/newfromtemplate.js

```diff
@@ -355,6 +367,15 @@ angular.module('openshiftConsole')
             $scope.parameterDisplayNames[parameter.name] = parameter.displayName || parameter.name;
           });
 
+          if($routeParams.templateParamsMap) {
+            var templateParams = getValidTemplateParamsMap();
+            _.each($scope.template.parameters, function(parameter) {
+              if (templateParams[parameter.name]) {
+                parameter.value = templateParams[parameter.name];
```

Comment:
Not sure I (we) understand the question.  $scope.template.parameters is the existing array that was already in use on the page.